src/GeoJson/Feature/Feature.php

```diff
@@ -21,9 +21,11 @@ class Feature extends GeoJson
     protected $geometry;
 
     /**
-     * Properties are a JSON object, which corresponds to an associative array.
+     * Properties are a JSON object, which corresponds to an associative array, or null.
      *
-     * @var array
+     * @see https://www.rfc-editor.org/rfc/rfc7946#section-3.2
```

Comment:
Yes, that's the section! When I ran the tests after updating the dependencies (the first commit), these were the relevant errors:

<details>
<summary>CLI Output</summary>

```
❯ php -v
PHP 8.1.1 (cli) (built: Dec 15 2021 09:54:28) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.1, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.1, Copyright (c), by Zend Technologies
    with Xdebug v3.1.2, Copyright (c) 2002-2021, by Derick Rethans

❯ phpunit
PHPUnit 9.5.11 by Sebastian Bergmann and contributors.

[...]

There were 4 errors:

1) GeoJson\Tests\Feature\FeatureCollectionTest::testUnserialization with data set "assoc=true" (true)
TypeError: GeoJson\Feature\Feature::__construct(): Argument #2 ($properties) must be of type array, null given

/Users/jg/Code/beste/geojson/src/GeoJson/Feature/Feature.php:43
/Users/jg/Code/beste/geojson/src/GeoJson/GeoJson.php:171
/Users/jg/Code/beste/geojson/src/GeoJson/GeoJson.php:141
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/Feature/FeatureCollectionTest.php:125
/Users/jg/Code/beste/geojson/vendor/phpunit/phpunit/phpunit:98

2) GeoJson\Tests\Feature\FeatureCollectionTest::testUnserialization with data set "assoc=false" (false)
TypeError: GeoJson\Feature\Feature::__construct(): Argument #2 ($properties) must be of type array, null given

/Users/jg/Code/beste/geojson/src/GeoJson/Feature/Feature.php:43
/Users/jg/Code/beste/geojson/src/GeoJson/GeoJson.php:171
/Users/jg/Code/beste/geojson/src/GeoJson/GeoJson.php:141
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/Feature/FeatureCollectionTest.php:125
/Users/jg/Code/beste/geojson/vendor/phpunit/phpunit/phpunit:98

3) GeoJson\Tests\Feature\FeatureTest::testConstructorShouldScanExtraArgumentsForCrsAndBoundingBox
TypeError: GeoJson\Feature\Feature::__construct(): Argument #2 ($properties) must be of type array, null given

/Users/jg/Code/beste/geojson/src/GeoJson/Feature/Feature.php:43
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/Feature/FeatureTest.php:15
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/BaseGeoJsonTest.php:16
/Users/jg/Code/beste/geojson/vendor/phpunit/phpunit/phpunit:98

4) GeoJson\Tests\Feature\FeatureTest::testSerializationWithCrsAndBoundingBox
TypeError: GeoJson\Feature\Feature::__construct(): Argument #2 ($properties) must be of type array, null given

/Users/jg/Code/beste/geojson/src/GeoJson/Feature/Feature.php:43
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/Feature/FeatureTest.php:15
/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/BaseGeoJsonTest.php:55
/Users/jg/Code/beste/geojson/vendor/phpunit/phpunit/phpunit:98

--

There was 1 failure:

1) GeoJson\Tests\Feature\FeatureTest::testSerializationWithNullConstructorArguments
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
 Array &0 (
     'type' => 'Feature'
     'geometry' => null
-    'properties' => null
+    'properties' => stdClass Object &00000000000003a50000000000000000 ()
 )

/Users/jg/Code/beste/geojson/tests/GeoJson/Tests/Feature/FeatureTest.php:60
/Users/jg/Code/beste/geojson/vendor/phpunit/phpunit/phpunit:98

ERRORS!
Tests: 128, Assertions: 389, Errors: 4, Failures: 1.
```
</details>

I first changed the signature so that it would allow `null`:

```diff
- public function __construct(Geometry $geometry = null, array $properties = array(), $id = null)
+ public function __construct(Geometry $geometry = null, ?array $properties = array(), $id = null)
``` 

That fixed the 4 errors, but the failure remained - with the properties defaulting to an empty array, `jsonSerialize()` converts it to a `stdClass`

https://github.com/jmikola/geojson/blob/980408cc70cd7d5f043f917d827dcc40d7470202/src/GeoJson/Feature/Feature.php#L95-L97

We could, of course, replace the `new \stdClass` with `null`, but I thought the change in #24 shouldn't have broken the tests in the first place 😅.

Reply:
I quickly created a branch with only those commits that update the dependencies and move to GH actions, so that you can see it in… action - apparently, the problem doesn't just occur in PHP 8.1 🙈 

https://github.com/beste/jmikola-geojson/runs/4877118084